### DISCUSSION TITLE
Propagate locale to nested relations

### DISCUSSION THREAD

**Discussion by Kinbaum** [directus/directus#7034] [Feature Requests]
Hi @benhaynes. First of all, thank you for this incredibly easy tool to use. It's been a breeze getting something up and running quickly.

I have a topic I'd like to discuss when it comes to translations. I have put together a `pages` collection that:
- Has some static fields
- Has it's own translatable fields
- Contains a M2O relationship (`some_section`)

Now within the `some_section` collection, this too has translatable fields as well as a M2A relationship.
My idea here is more focused on the retrieval of the data rather than the interface for creating the translations. The proposed design [here](https://github.com/directus/directus/discussions/3011#discussioncomment-706937) is actually really nice, hope to see this soon!

### The Idea

Consider the following GraphQL example:

```gql
query Pages($language: String!) {
    pages {
        static_field_1
        static_field_2
        translations(
            filter: { languages_code: { code: { _starts_with: $language } } }
        ) {
            title
            description
        }
        some_section {
            translations(
                filter: {
                    languages_code: { code: { _starts_with: $language } }
                }
            ) {
                title
                description
            }
            embedded_section {
                item {
                    ... on some_component {
                        translations(
                            filter: {
                                languages_code: { code: { _starts_with: $language } }
                            }
                        ) {
                            title
                            description
                        }
                    }
                }
            }
        }
    }
}
```

You can see that based on the example above, I've parameterized the query so that the language can be passed to all translatable collections. I find that having to do this to be a bit too repetitive and there should be a better way of handling the translations.

If this is already possible to do and I'm missing it, please forgive me; ideally I'd like to be able to set the context of the query by passing a `locale` at the root, and have that propagate to the nested relationships. This would clean up the query and remove the duplication. Here is what my ideal query would look like:

```gql
query Pages($locale: String!) {
    pages(locale: $locale) {
        static_field_1
        static_field_2
        title
        description
        some_section {
            title
            description
            embedded_section {
                item {
                    ... on some_component {
                        title
                        description
                    }
                }
            }
        }
    }
}
```

Please let me know your thoughts on this. 
Thanks

**Comment by rijkvanzanten** [comment]
This is the same as https://github.com/directus/directus/discussions/6654 🙂

**Reply by Kinbaum** [reply]
Great, thank you. Thumbs up on that one

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
